src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs

```diff
@@ -117,24 +118,28 @@ public virtual int GetInt32(int ordinal)
 
         public virtual long GetInt64(int ordinal)
             => IsDBNull(ordinal)
-                ? throw new InvalidCastException()
+                ? throw new InvalidOperationException(Resources.CalledOnNullValue)
                 : GetInt64Core(ordinal);
 
         protected abstract long GetInt64Core(int ordinal);
 
         public virtual string GetString(int ordinal)
             => IsDBNull(ordinal)
-                ? throw new InvalidCastException()
+                ? throw new InvalidOperationException(Resources.CalledOnNullValue)
                 : GetStringCore(ordinal);
 
         protected abstract string GetStringCore(int ordinal);
 
         public virtual T GetFieldValue<T>(int ordinal)
+            => GetFieldValue<T>(ordinal, Resources.CalledOnNullValue);
+
+        public virtual T GetFieldValue<T>(int ordinal, string onNullErrorMsg)
```

Comment:
Callers should have to pass in the message. Can we simply override it (via an protected abstract property or something) in `SqliteDataRecord` and `SqliteParamterReader`? You can flow the function name in through the `SqliteParameterReader` constructor.